module/applications/advancement/ability-score-improvement-flow.mjs

```diff
@@ -57,7 +60,7 @@ export default class AbilityScoreImprovementFlow extends AdvancementFlow {
     const abilities = Object.entries(CONFIG.DND5E.abilities).reduce((obj, [key, data]) => {
       if ( !this.advancement.canImprove(key) ) return obj;
       const ability = this.advancement.actor.system.abilities[key];
-      const fixed = this.advancement.configuration.fixed[key] ?? 0;
+      const fixed = (this.advancement.configuration.fixed[key] || ability.max === ability.value) ?? 0;
```

Comment:
The result of this expression will sometimes be a boolean, but `fixed` must always be a number because it's used in an arithmetic expression on the following line.

I think it would be better if we change `isFixed` on line 72 (75 in this PR) to:
```js
isFixed: !!fixed || (ability.value >= ability.max)
```